pandas/core/strings.py

```diff
@@ -2389,6 +2388,15 @@ def cat(self, others=None, sep=None, na_rep=None, join="left"):
         split : Split each string in the Series/Index.
         join : Join lists contained as elements in the Series/Index.
 
+        Notes
+        -----
+        Since version 1.0.0, index alignment is performed when
+        `others` is a Series/Index/DataFrame (or a list-like containing
```

Comment:
is this true for an Index?  If the behavior is anything like arithmetic, we would do alignment for Series/DataFrame, but Index would behave more like ndarray